packages/sdk/src/util/handle-response.ts

```diff
@@ -0,0 +1,24 @@
+export async function handleResponse(response: Response, contextMessage?: string): Promise<any> {
```

Comment:
In this method I would do the following changes:
- Since there might be the case where the response won't come in JSON format and we can then get the error of "not being able to parse this JSON" if the `await response.json()` is done 1st, I would place the `if (!response.ok) {...` block 1st at the top.
- Remove the `data` error handling you have bellow from there, and place it inside the `try` since it's there you have the `data` object. Meaning this block:
```
    if (data.error && data.error.message) {
      errorMsg += `. Error message: ${data.error.message}`;
    } else if (data.message) {
      errorMsg += `. Error message: ${data.message}`;
    }
```
- Then add the `return data;` inside the try block as well